internal/pkg/cli/env_delete.go

```diff
@@ -291,7 +301,26 @@ func (o *deleteEnvOpts) validateNoRunningServices() error {
 				svcNames = append(svcNames, *t.Value)
 			}
 		}
-		return fmt.Errorf("service '%s' still exist within the environment %s", strings.Join(svcNames, ", "), o.name)
+		return fmt.Errorf("service %q still exist within the environment %s", strings.Join(svcNames, ", "), o.name)
+	}
+	return nil
+}
+
+func (o *deleteEnvOpts) validateNoDependencyPipelines() error {
+	pipelines, err := o.deployedPipelineLister.ListDeployedPipelines(o.appName)
+	if err != nil {
+		return fmt.Errorf("list deployed pipelines: %w", err)
+	}
+	for _, pipeline := range pipelines {
+		info, err := o.pipelineGetter.GetPipeline(pipeline.ResourceName)
+		if err != nil {
+			return fmt.Errorf("get pipeline %s: %w", pipeline.ResourceName, err)
+		}
+		for _, stage := range info.Stages {
+			if strings.TrimPrefix(stage.Name, deploy.StageFullNamePrefix) == o.name {
```

Comment:
Is this the most durable way we can find this out? Do we stick the manifest in the pipeline metadata?

Reply:
That's a very good question. We've never changed our naming pattern for this and I am making it more durable by extracting `deploy.StageFullNamePrefix`.

>Do we stick the manifest in the pipeline metadata?

Unfortunately we don't do that for manifest :(